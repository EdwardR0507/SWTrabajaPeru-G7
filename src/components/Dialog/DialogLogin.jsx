import React from 'react';
import { useHistory } from 'react-router';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';

const styles = (theme) => ({
    root: {
        margin: 0,
        padding: theme.spacing(2),
    },
    closeButton: {
        position: 'absolute',
        right: theme.spacing(1),
        top: theme.spacing(1),
        color: theme.palette.grey[500],
    },
});

const DialogTitle = withStyles(styles)((props) => {
    const { children, classes, onClose, ...other } = props;
    return (
        <MuiDialogTitle disableTypography className={classes.root} {...other}>
            <Typography variant="h6">{children}</Typography>
            {onClose ? (
                <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
                    <CloseIcon />
                </IconButton>
            ) : null}
        </MuiDialogTitle>
    );
});

const DialogContent = withStyles((theme) => ({
    root: {
        padding: theme.spacing(2),
    },
}))(MuiDialogContent);

const DialogActions = withStyles((theme) => ({
    root: {
        margin: 0,
        padding: theme.spacing(1),
    },
}))(MuiDialogActions);

export default function DialogLogin() {
    const [open, setOpen] = React.useState(false);
    const history = useHistory();

    const handleClickOpen = () => {
        setOpen(true);
    };
    const handleClose = () => {
        setOpen(false);
    };

    return (
        <div>
            <Button color="primary" onClick={handleClickOpen}>
                Ver M치s
            </Button>
            <Dialog onClose={handleClose} aria-labelledby="customized-dialog-title" open={open}>
                <DialogContent>
                    <Typography gutterBottom>
                        Para poder acceder a esta informaci칩n usted debe iniciar sesi칩n.
                    </Typography>
                </DialogContent>
                <DialogActions>
                    <Button autoFocus onClick={()=>{
                        history.push({
                            pathname: "/signin"
                        })
                    }} color="primary" variant="contained">Iniciar Sesi칩n</Button>
                    <Button autoFocus onClick={handleClose} color="secondary" variant="contained" >
                        Cancelar
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
}