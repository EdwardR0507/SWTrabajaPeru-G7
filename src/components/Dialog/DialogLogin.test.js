/*Importamos las librerias principales*/
import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks'
import { shallow, mount } from "enzyme";
import DialogLogin from "./DialogLogin";

describe("<DialogLogin />", () => {
  it("render DialogLogin", () => {
    const wrapper = shallow(<DialogLogin />);
    expect(wrapper).toMatchSnapshot()
  })
  /*it("test handle open", () => {
      const { result, getByRole } = render(
          <DialogLogin />
      );
      fireEvent.click(getByRole("open"));
      expect(result.current.open).toBe(true);
  })
  it("test handle close", () => {
      jest.mock('react-router-dom', () => ({
          ...jest.requireActual('react-router-dom'),
          useHistory: () => ({
            push: mockHistoryPush, 
          }),
        }));
      const { result, getByRole } = renderHook(
          <DialogLogin />
      );
      fireEvent.click(getByRole("close"));
      expect(result.current.open).toBe(false);
  })*/
  it('Redirects to correct URL on click', () => {
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useHistory: () => ({
        push: mockHistoryPush,
      }),
    }));
    const wrapper = shallow(<DialogLogin />);
    const button = wrapper.find({ role: "redirect" });
    button.simulate("click");
    expect(mockHistoryPush).toHaveBeenCalledWith('/signin');
  });
  it('handleClose test', () => {
    const wrapper = shallow(<DialogLogin />);
    const button = wrapper.find({ role: "open-button" });
    button.simulate("click");
    expect(wrapper.state().open).toBe(true)
  })
  it('handleOpen test', () => {
    const wrapper = shallow(<DialogLogin />);
    const button = wrapper.find({ role: "close-button" });
    button.simulate("click");
    expect(wrapper.state().open).toBe(false)
  })
})