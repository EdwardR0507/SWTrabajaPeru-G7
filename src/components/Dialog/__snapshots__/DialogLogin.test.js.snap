// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogLogin /> render DialogLogin 1`] = `ShallowWrapper {}`;
